python/nano/notebooks/nano-inference-example.ipynb

```diff
@@ -0,0 +1,400 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "seed_everything(7)\n",
+    "PATH_DATASETS = os.environ.get(\"PATH_DATASETS\", \".\")\n",
+    "BATCH_SIZE = 64\n",
+    "NUM_WORKERS = int(os.cpu_count() / 2)\n",
+    "data_module = prepare_data(PATH_DATASETS, BATCH_SIZE, NUM_WORKERS)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b7d4b74a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pl_model = LitResnet(learning_rate=0.05)\n",
+    "pl_model.datamodule = data_module\n",
+    "trainer = Trainer(num_processes = 1,\n",
+    "                  use_ipex = False,\n",
+    "                  progress_bar_refresh_rate=10,\n",
+    "                  max_epochs=30,\n",
+    "                  logger=TensorBoardLogger(\"lightning_logs/\", name=\"basic\"),\n",
+    "                  callbacks=[LearningRateMonitor(logging_interval=\"step\")])\n",
+    "trainer.fit(pl_model, datamodule=data_module)\n",
+    "trainer.test(pl_model, datamodule=data_module)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "985d681d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pl_model = LitResnet.load_from_checkpoint('lightning_logs/resnet.ckpt', )\n",
+    "data_module.setup(\"test\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3c30c24e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start = time()\n",
+    "pl_model.eval()\n",
+    "for x, _ in data_module.test_dataloader():\n",
+    "    with torch.no_grad():\n",
+    "        pl_model(x)\n",
+    "infer_time = time() - start\n",
+    "outputs = trainer.test(pl_model, datamodule=data_module)\n",
+    "fp32_acc = outputs[0]['test_acc'] * 100"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01c93f5f",
+   "metadata": {},
+   "source": [
+    "### Get Accelerated Module\n",
+    "---\n",
+    "Use Train.trace from bigdl.nano.pytorch.trainer to convert a model into an accelerated module for inference.\n",
+    "The definition of trace is:\n",
+    "```\n",
+    "trace(model: nn.Module, input_sample=None, accelerator=None)\n",
+    "\n",
+    "      :param model: An torch.nn.Module model, including pl.LightningModule.\n",
+    "      \n",
+    "      :param input_sample: A set of inputs for trace, defaults to None if you have trace before or\n",
+    "                             model is a LightningModule with an example_input_array.\n",
+    "                             \n",
+    "      :param accelerator: The accelerator to use, defaults to None meaning staying in Pytorch\n",
+    "                            backend. 'openvino' and 'onnxruntime' are supported for now.\n",
+    "                            \n",
+    "      :return: Model with different acceleration(OpenVINO/ONNX Runtime).\n",
+    "```\n",
+    "- *Note* <br>\n",
+    "trace is a class method. You should use your Trainer class to call it instead of the Trainer instance."
```

Comment:
I think it is also OK to call it using the Trainer instance, right? @TheaperDeng @zhentaocc

Reply:
Yes.